src/main/java/HM04/Gold.java

```diff
@@ -0,0 +1,20 @@
+package HM04;
+
+public class Gold implements BasicInterface {
+
+    private int quantity;
```

Comment:
You have this field and getter and same constructor in all implemantations. In this case good reason use abstract class with this logic